src/cow-react/modules/limitOrders/containers/TradeButtons/index.tsx

```diff
@@ -6,28 +6,45 @@ import { tradeFlow, TradeFlowContext } from '@cow/modules/limitOrders/services/t
 import { limitOrdersSettingsAtom } from '@cow/modules/limitOrders/state/limitOrdersSettingsAtom'
 import { useLimitOrdersTradeState } from '@cow/modules/limitOrders/hooks/useLimitOrdersTradeState'
 import { useLimitOrdersFormState } from '../../hooks/useLimitOrdersFormState'
-import { limitOrdersTradeButtonsMap, SwapButton } from './limitOrdersTradeButtonsMap'
+import { limitOrdersTradeButtonsMap, SwapButton, WrapUnwrapParams } from './limitOrdersTradeButtonsMap'
 import { limitOrdersConfirmState } from '../LimitOrdersConfirmModal/state'
-import { useToggleWalletModal } from 'state/application/hooks'
+import { useCloseModals, useModalIsOpen, useToggleWalletModal } from 'state/application/hooks'
 import { limitOrdersQuoteAtom } from '@cow/modules/limitOrders/state/limitOrdersQuoteAtom'
 import { useLimitOrdersWarningsAccepted } from '@cow/modules/limitOrders/hooks/useLimitOrdersWarningsAccepted'
 import { PriceImpact } from 'hooks/usePriceImpact'
+import { useWrapCallback } from 'hooks/useWrapCallback'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { transactionConfirmAtom } from '@cow/modules/swap/state/transactionConfirmAtom'
+import { ApplicationModal } from '@src/state/application/reducer'
 
 export interface TradeButtonsProps {
   tradeContext: TradeFlowContext | null
   priceImpact: PriceImpact
+  inputCurrencyAmount: CurrencyAmount<Currency> | null
   openConfirmScreen(): void
 }
 
 export function TradeButtons(props: TradeButtonsProps) {
-  const { tradeContext, openConfirmScreen, priceImpact } = props
+  const { tradeContext, openConfirmScreen, priceImpact, inputCurrencyAmount } = props
   const { expertMode } = useAtomValue(limitOrdersSettingsAtom)
   const formState = useLimitOrdersFormState()
   const tradeState = useLimitOrdersTradeState()
   const setConfirmationState = useSetAtom(limitOrdersConfirmState)
   const toggleWalletModal = useToggleWalletModal()
   const quote = useAtomValue(limitOrdersQuoteAtom)
   const warningsAccepted = useLimitOrdersWarningsAccepted(false)
+  const wrapUnwrapCallback = useWrapCallback(inputCurrencyAmount || undefined)
+  const transactionConfirmState = useAtomValue(transactionConfirmAtom)
+  const closeModals = useCloseModals()
+  const showTransactionConfirmationModal = useModalIsOpen(ApplicationModal.TRANSACTION_CONFIRMATION)
+
+  const wrapUnwrapParams: WrapUnwrapParams = {
+    isNativeIn: !!inputCurrencyAmount?.currency.isNative,
```

Comment:
It might be overkill, but there's a hook that returns part of this info https://github.com/cowprotocol/cowswap/blob/ade1ecbfc64e6820a95f6479f8768c7bbb44c66a/src/cow-react/modules/swap/hooks/useDetectNativeToken.ts#L41

Reply:
Yep, I found it too complex to use `useDetectNativeToken`